IoTPiCar.py

```diff
@@ -41,163 +55,135 @@ def calculate_cycles_from_distance(distance: int):
+#         print('detection dist', detection_distances)
+#         safe_forward_cycles = calculate_cycles_from_distance(max_safe_distance)
+#         print('safe forward cyucles', safe_forward_cycles)
+#         if(max_safe_distance > 0 and max_safe_distance <= 80):
+#             print('moving forward')
+#             #move(safe_forward_cycles, movement_type='linear', direction='forward')
+
+def create_advanced_mapping(sweep_info):
+    global envoriment_map
+    for entry in sweep_info:
+        if(entry['detection']):
+            #(x,y) with respect to the car
+            scanned_object_x = entry['distance']*math.sin(math.radians(entry['angle']))
+            scanned_object_y = entry['distance']*math.cos(math.radians(entry['angle']))
+                        
+            #(x,y) with respect to the ma           
+            map_x = car_start_map_x + scanned_object_x
+            map_y = car_start_map_y + scanned_object_y
             
-    if(not isSomethingAhead):
-        max_safe_distance = min(detection_distances) -THIRTY_FIVE_CM
-        print('detection dist', detection_distances)
-        safe_forward_cycles = calculate_cycles_from_distance(max_safe_distance)
-        print('safe forward cyucles', safe_forward_cycles)
-        if(max_safe_distance > 0 and max_safe_distance <= 80):
-            print('moving forward')
-            #move(safe_forward_cycles, movement_type='linear', direction='forward')
+            if((map_x >= MAP_SIZE) or (map_y >= MAP_SIZE)):
+                print('x:',scanned_object_x, 'y:',scanned_object_y,'xx:',map_x,'yy:',map_y)
+            else:
+                envoriment_map[int(map_x), int(map_y)] = entry['detection']
         
-
 if __name__ == '__main__':
-    # self_drive()
-    #servo.set_angle(0)
-    # #move(5, movement_type='rotation', direction='left')
-    # distance = ultrasonic.get_distance()
-    # if(distance > 35):
-    #     picar.forward
-    # picar.stop()
-    # print()
-    isSafeToMove = True
-    while isSafeToMove:
-        picar.forward(WHEEL_POWER)
-        isSafeToMove = not perform_moving_scan(servo_speed=0.02)
-        if(not isSafeToMove):
-            picar.stop()
-            stationary_scan = perform_stationary_scan()
-            decide_movement(stationary_scan)
-    picar.stop()
+    try:
+        set_servo_angle(SERVO_MIN_ANGLE)
+        
+        #Camera setting Camera Id = 0
+        camera = cv.VideoCapture(0)
+        camera.set(cv.CAP_PROP_FRAME_WIDTH, 640)
+        camera.set(cv.CAP_PROP_FRAME_HEIGHT, 480)
+         
+        #Tensorflow model settings
+        tensorflow_model_options = ObjectDetectorOptions(
+            num_threads=4,
+            score_threshold=0.5,
+            max_results=3,
+            enable_edgetpu=False)
+         
+        detector = ObjectDetector(model_path='efficientdet_lite0.tflite', options=tensorflow_model_options)
+
+        #While ESC key is not pressed
+        while (cv.waitKey(1) != 27):           
```

Comment:
could we pack this condition into a variable to avoid code comments.

Reply:
sure we could if that gives more clarity but here isnt this more clear?